.github/workflows/ci.yaml

```diff
@@ -597,14 +597,14 @@ jobs:
         run: |
           . venv/bin/activate
           python --version
-          pylint --ignore-missing-annotations=y homeassistant
+          pylint --ignore-missing-annotations=y homeassistant --exit-zero --fail-on=W,E,F
```

Comment:
I think we need to review what messages no longer will fail CI in the convention and refactor category.

https://pylint.readthedocs.io/en/stable/user_guide/messages/messages_overview.html

I think many in those two categories are handled by ruff and already disabled in pylint but I don't know which exactly.

In the information category it's only "use-symbolic-message-instead" that is currently enabled and relevant.

https://pylint.readthedocs.io/en/stable/user_guide/messages/information/use-symbolic-message-instead.html

Reply:
Hm I think it is better to have a option like `ignore-missing-annotations`, which is enabled in CI but not locally. This way everything else is unaffected

Reply:
Sounds good!